src/Surfnet/ServiceProviderDashboard/Application/Command/Entity/SaveSamlEntityCommand.php

```diff
@@ -106,6 +107,19 @@ class SaveSamlEntityCommand implements SaveEntityCommandInterface
     #[Assert\Url]
     private ?string $applicationUrl = null;
 
+    /** @var TypeOfService[] */
+    #[Assert\All([
+        new Assert\NotBlank(),
+        new Assert\Type(type: TypeOfService::class),
+    ])]
+    #[Assert\Count(
+        min: 1,
+        max: 3,
+        minMessage: 'validator.type-of-service.min',
+        maxMessage: 'validator.type-of-service.max',
+    )]
+    private array $typeOfService;
+
     #[Assert\Url]
```

Comment:
Do I see some repetition here? DRY?

Reply:
You are looking at a (long) form with several elements that have similar characteristics. They however represent entities of their own. I'm considering making a custom type of service validator. Suitable for the 3 entity types.